datarepo-clienttests/src/main/java/scripts/testscripts/SnapshotScaleDelete.java

```diff
@@ -0,0 +1,106 @@
+package scripts.testscripts;
+
+import bio.terra.datarepo.api.RepositoryApi;
+import bio.terra.datarepo.client.ApiClient;
+import bio.terra.datarepo.model.DeleteResponseModel;
+import bio.terra.datarepo.model.JobModel;
+import bio.terra.datarepo.model.SnapshotModel;
+import bio.terra.datarepo.model.SnapshotSummaryModel;
+import com.google.cloud.storage.BlobId;
+import common.utils.StorageUtils;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import runner.config.TestUserSpecification;
+import scripts.testscripts.baseclasses.SimpleDataset;
+import scripts.utils.DataRepoUtils;
+
+public class SnapshotScaleDelete extends SimpleDataset {
+  private static final Logger logger = LoggerFactory.getLogger(SnapshotScaleDelete.class);
+
+  /** Public constructor so that this class can be instantiated via reflection. */
+  public SnapshotScaleDelete() {
+    super();
+  }
+
+  private SnapshotModel snapshotModel;
+
+  private static List<BlobId> scratchFiles = new ArrayList<>();
+
+  private int NUM_SNAPSHOTS = 1;
+
+  public void setParameters(List<String> parameters) {
+    if (parameters != null && parameters.size() > 0) {
+      NUM_SNAPSHOTS = Integer.parseInt(parameters.get(0));
+    }
+    logger.debug("Number of snapshots to create and delete (default is 1): {}", NUM_SNAPSHOTS);
+  }
+
+  public void setup(List<TestUserSpecification> testUsers) throws Exception {
+    // create the profile and dataset
+    super.setup(testUsers);
+  }
+
+  public void userJourney(TestUserSpecification testUser) throws Exception {
+    ApiClient apiClient = DataRepoUtils.getClientForTestUser(testUser, server);
```

Comment:
I think this needs to be "datasetCreator" instead of testUser in order to guarantee you're using the same user throughout the test.

Reply:
Or maybe not. Things fail for me when I set this, so IDK anymore.